main.go

```diff
@@ -150,6 +153,14 @@ func main() {
 	ctx := ctrl.SetupSignalHandler()
 	setupIndexes(ctx, mgr, &cfg)
 
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeConfig)
+	if err != nil {
+		setupLog.Error(err, "Unable to discovery client")
+		os.Exit(1)
+	}
+
+	go discoveryversion.FetchServerVersion(ctx, discoveryClient)
```

Comment:
Should we register this to manager like scheduler?

Reply:
Yeah, it would be nice to have a generic interface such as `KubeVersionFetcher`. It could be then initialized as a caching fetcher in prod, but as a fetcher returning constant in tests. I'm not sure this is required at this point though. wdyt @alculquicondor ?